stress_test/nb_generator.py

```diff
@@ -0,0 +1,278 @@
+
+""" NB-Generator Class- All NB-Generator-related functionality is here"""
+
+import gevent
+import logging
+import time
+import util.netutil
+
+
+class NBgen:
+
+    def __init__(self, nb_gen_base_dir, test_config, controller, sbemu,
+                 log_level="DEBUG"):
+        """Create an NB-generator object. Options from JSON input file
+        :param test_config: JSON input configuration
+        :param nb_gen_base_dir: emulator base directory
+        :param controller: object of the Controller class
+        :param sbemu: object of the SBEmu subclass
+        :type test_config: JSON configuration dictionary
+        :type nb_gen_base_dir: str
+        :type controller: object
+        :type sbemu: object
+        """
+        self.controller = controller
+        self.sbemu = sbemu
+        self.name = test_config['nb_generator_name']
+        self.base_dir = nb_gen_base_dir
+
+        self.ip = test_config['nb_generator_node_ip']
+        self.ssh_port = test_config['nb_generator_node_ssh_port']
+        self.ssh_user = test_config['nb_generator_node_username']
+        self.ssh_pass = test_config['nb_generator_node_password']
+
+        self.run_hnd = (self.base_dir +
+                        test_config['nb_generator_run_handler'])
+        self.status = 'UNKNOWN'
+        self._ssh_conn = None
+
+        self.flow_delete_flag = test_config['flow_delete_flag']
+        self.flows_per_request = test_config['flows_per_request']
+        self.log_level = log_level
+
+        # The parameters initialized as None are dimensions of the test.
+        # These values are passed outside, from the test in the main for loop.
+        # ---------------------------------------------------------------------
+        self.flow_workers = None
+        self.total_flows = None
+        self.flow_operations_delay_ms = None
+        # ---------------------------------------------------------------------
+        self.flows_ds_discovery_deadline = 240
+
+        self.confirm_time = 0.0
+        self.e2e_installation_time = 0.0
+        self.discover_flows_on_switches_time = 0.0
+
+    def init_ssh(self):
+        logging.info(
+            '[open_ssh_connection] Initiating SSH session with {0} node.'.
+            format(self.name, self.ip))
+
+        self._ssh_conn = \
+            util.netutil.ssh_connect_or_return2(self.ip,
+                                                int(self.ssh_port),
+                                                self.ssh_user,
+                                                self.ssh_pass,
+                                                10)
+
+    def run(self):
+        """ Wrapper to the NB-Generator run handler
+        :returns: Returns the combined stdout - stderr of the executed command
+        :rtype: str
+        """
+        logging.info("[NB_generator] Run handler")
+        self.status = 'STARTED'
+        cmd = ('cd {0}; python3.4 {1} {2} {3} {4} {5} {6} {7} {8} '
```

Comment:
Add bash meta handler for running northbound  generator. Purpose of the meta handler to activate virtual environment having all dependencies of northbound generator. See meta-handler from Multinet.